java/ql/src/semmle/code/java/dataflow/internal/DataFlowUtil.qll

```diff
@@ -84,6 +84,17 @@ class Node extends TNode {
     result = this.(ImplicitPostUpdateNode).getPreUpdateNode().getEnclosingCallable()
   }
 
+  /** Gets the callable in which this node occurs. */
+  DataFlowBasicBlock getBasicBlock() {
+    // TODO check all cases!
+    result = this.asExpr().getBasicBlock() or
+    //result = this.asParameter().getAnAccess().getBasicBlock() or
+    //result = this.(ImplicitVarargsArray).getCall() or
+    result = this.(ImplicitInstanceAccess).getInstanceAccess().getCfgNode().getBasicBlock() or
+    result = this.(MallocNode).getClassInstanceExpr().getBasicBlock()
+    //result = this.(ImplicitPostUpdateNode).getBasicBlock()
+  }
```

Comment:
Come to think of it, I think this should be moved out as a top-level predicate in `DataFlowPrivate.qll` instead, in order not to pollute the public API. Also, the QL doc needs to be updated.